### DISCUSSION TITLE
Cannot select or include count in typescript

### DISCUSSION THREAD

**Discussion by IrisMurray** [prisma/prisma#9324] [Q&A]
I have a `Location` table that has a 1-N relationship with `Event` and each `Event` has a M-1 relation with `People`.
I'd eventually like to be able to count the people for a particular location.
I followed the count relation instructions at [https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#count-relations](https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#count-relations)
However as a go between I first tried to count the events for a location with id:1 using
```ts
prisma.location.findUnique({
  where: {id:1}, 
  select: {
    _count: {
      select: {
        event:true
      }
    }
  }
})
```
But I get an error Type '{ _count: { select: { event: true; }; }; }' is not assignable to type 'LocationSelect'.
 Object literal may only specify known properties, and '_count' does not exist in type 'LocationSelect'.ts(2322)

Any hints?

**Comment by ryands17** [comment]
@IrisMurray 👋 

Works fine for me. What version of `prisma` and `@prisma/client` are you on?

Also for now you can only count for the first relation and not nested ones. You can only find the count of events for a given location, not the count of people for a given location as people is not directly related to location.

**Reply by IrisMurray** [reply]
I'm on `prisma@2.30.3` and `@prisma/client@2.30.3`

My schema, with non-relevant fields removed, is:
```prisma
model Location{
id    Int    @id @default(autoincrement())
events    Event[]
}

model Event{
id    Int    @id @default(autoincrement())
location    Location    @relation(fields:[locationId], references[id])
person    Person     @relation(fields:[personId], references[id])
locationId    Int
personId    Int
}

model Person{
id    Int    @id @default(autoincrement())
events    Event[] 
}
```

My code that is trying to call the defined function is as:
```ts
async getEventCount(id: number){
return await  prisma.location.findUnique({
  where: {id}, 
  select: {
    _count: {
      select: {
        events:true
      }
    }
  }
})
}
```

Looking through the generated `.prisma/client` files (`index.d.ts`) `location.findUnique` has type
```
    findUnique<T extends LocationFindUniqueArgs,  LocalRejectSettings = T["rejectOnNotFound"] extends RejectOnNotFound ? T['rejectOnNotFound'] : undefined>(
      args: SelectSubset<T, LocationFindUniqueArgs>
    ): HasReject<GlobalRejectSettings, LocalRejectSettings, 'findUnique', 'Location'> extends True ? CheckSelect<T, Prisma__LocationClient<Location>, Prisma__LocationClient<LocationGetPayload<T>>> : CheckSelect<T, Prisma__LocationClient<Location | null >, Prisma__LocationClient<LocationGetPayload<T> | null >>
```

 `LocationFindUniqueArgs` has type definition
```ts
  export type LocationFindUniqueArgs = {
    select?: LocationSelect | null
    include?: LocationInclude | null
    rejectOnNotFound?: RejectOnNotFound
    where: LocationWhereUniqueInput
  }
```
and `LocationSelect` has type definition
```ts
  export type LocationSelect = {
    id?: boolean
    events?: boolean | EventFindManyArgs
  }
```

I understand that I should only currently be able to count the first relation (which is also useful in my particular use case)

**Reply by ryands17** [reply]
It's strange that you don't have `_count` in `LocationSelect`. I unfortunately cannot reproduce your issue and here is a working example.

[test-prisma.zip](https://github.com/prisma/prisma/files/7200676/test-prisma.zip)

**Reply by IrisMurray** [reply]
We have a difference in version numbers of prisma and client. I could not do the count select with version `2.30.3`, but it has appeared when I have updated everything to version `3.1.1`. The `_count` field has magically  appeared in LocationSelect

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
